tasks/main.yml

```diff
@@ -28,6 +28,13 @@
     state: directory
   with_items: "{{ os_images_list }}"
 
+- name: Download the image to the directory when an image_url is defined
+  get_url:
+    url: "{{ item.image_url }}"
+    dest: "{{ os_images_cache }}/{{ item.name }}/{{ item.name }}.qcow2"
+  with_items: "{{ os_images_list }}"
+  when: item.image_url is defined
```

Comment:
How about registering the download result, and (in a similar manner to the builds, below) add a check to fail if any downloads were not successful.

Reply:
I just tested this with a dummy URL and looks like Ansible automatically fails if the URL is unreachable.